src/Simulation/EntryPointDriver/Azure/Azure.cs

```diff
@@ -418,7 +418,9 @@ private static void LogIfVerbose(GenerateAzurePayloadSettings settings, string m
             null => null,
             NoOpQirSubmitter.Target => new NoOpQirSubmitter(),
             NoOpSubmitter.Target => new NoOpSubmitter(),
-            _ => SubmitterFactory.QirSubmitter(settings.Target, settings.CreateWorkspace(), settings.Storage)
+            _ => (settings.TargetCapability != null) 
```

Comment:
`TargetCapability` can never be null.

Reply:
Fixed. Now the check is against white-space:
```c#
 _ => (!string.IsNullOrWhiteSpace(settings.TargetCapability)) 
```